app/certificates.go

```diff
@@ -0,0 +1,429 @@
+package app
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/temporalio/tcld/utils"
+	"github.com/urfave/cli/v2"
+)
+
+func generateRandomString(n int) (string, error) {
+	const letters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-"
+	ret := make([]byte, n)
+	for i := 0; i < n; i++ {
+		num, err := rand.Int(rand.Reader, big.NewInt(int64(len(letters))))
+		if err != nil {
+			return "", err
+		}
+		ret[i] = letters[num.Int64()]
+	}
+
+	return string(ret), nil
+}
+
+type generateCACertificateInput struct {
+	Organization string        `validate:"required"`
+	Duration     time.Duration `validate:"required"`
+	RSAAlgorithm bool
+}
+
+func generateCACertificate(
+	input generateCACertificateInput,
+) (caPEM, caPrivateKeyPEM []byte, err error) {
+	validator := validator.New()
+	if err := validator.Struct(input); err != nil {
+		return nil, nil, err
+	}
+	randomLetters, err := generateRandomString(4)
+	if err != nil {
+		return nil, nil, err
```

Comment:
[Nit] wrap error with more friendly error message for user (e.g. unable to generate random   string for dns name or something like that?)